packages/rum/src/domain/record/observers/recordIds.ts

```diff
@@ -0,0 +1,15 @@
+export type RecordIds = ReturnType<typeof initRecordIds>
+
+export function initRecordIds() {
+  const recordIds = new WeakMap<Event, number>()
+  let nextId = 1
+
+  return {
+    get(event: Event): number {
```

Comment:
Looking at it, it was not utility code but domain logic, so following the convention, I passed it as a dependency rather than statically import it.
By automatism, I switched the singleton getter to an object but indeed we could keep the singleton getter and pass it as a dependency.
Any pros/cons to favour the singleton?

Reply:
I'm a bit surprised to see this kind of change included in a PR focused on splitting files. I have mixed feelings on enforcing the "dependency as parameter" policy when it's not really needed, but fine.

Still, I think the `getRecordIdForEvent` name was clearer than `recordIds.get`. Maybe we could rename it `recordIds.getIdForEvent`?